cargo-near-build/src/near/build/mod.rs

```diff
@@ -15,18 +16,26 @@ use crate::{
             manifest_path::{ManifestPath, MANIFEST_FILE_NAME},
             metadata::CrateMetadata,
         },
-        near::build::{input::Opts, output::version_mismatch::VersionMismatch},
+        near::build::{
+            input::{self, Opts},
+            output::version_info::VersionInfo,
+        },
     },
 };
 
 use super::abi;
 
-pub mod export;
-
 /// builds a contract whose crate root is current workdir, or identified by [`Cargo.toml`/BuildOpts::manifest_path](crate::BuildOpts::manifest_path) location
-pub fn run(args: Opts) -> eyre::Result<CompilationArtifact> {
-    VersionMismatch::export_builder_and_near_abi_versions();
-    export::nep_330_build_command(&args)?;
+pub fn run(
+    args: Opts,
+    implicit_env_opts: Option<input::implicit_env::Opts>,
```

Comment:
Please, look at the public API from the usage point of view.

1. This (Opts, Option<Opts>) is really hard to follow.
2. `nep330_contract_path` doesn't make sense to me as a user of the package - I can understand `contract_path`, but I would never thought about it in terms of "Oh, I need to specify NEP330 parameters, here we go nep330_contract_path".